content/document.js

```diff
@@ -357,6 +357,9 @@ function move(oldSlug, newSlug, locale, { dry = false } = {}) {
   if (!doc) {
     throw new Error(`document for ${oldSlug} does not exist`);
   }
+  if (newSlug.includes("#")) {
```

Comment:
if we move this into the `saveHTMLFile` function we should catch many more ways of introducing an invalid slug.

Reply:
Good idea! I haven't tested if this breaks the importer. But I guess we can cross that bridge when we get there.